oam/install_scripts/columnstore-post-install.in

```diff
@@ -310,16 +310,13 @@ chmod 777 /dev/shm
 
 systemctl cat mariadb-columnstore.service > /dev/null 2>&1
 if [ $? -eq 0 ] && [ $(running_systemd) -eq 0 ]; then
-    # prevent clusters using shared storage from initializing columnstore more than once
-    IFLAG=/var/lib/columnstore/storagemanager/cs-initialized
     mkdir -p /var/lib/columnstore/storagemanager
     chown -R mysql:mysql /var/lib/columnstore/storagemanager
 
-    if [ ! -e $IFLAG ]; then
-       touch $IFLAG
-       echo "Populating the engine initial system catalog."
-       systemctl start mariadb-columnstore
-    fi
+    echo "Populating the engine initial system catalog."
+
+    # prevent nodes using shared storage manager from stepping on each other when initializing
+    flock /var/lib/columnstore/storagemanager -c "systemctl start mariadb-columnstore"
```

Comment:
I wonder if flock must defend against multiple runs of the command on the same node?

Reply:
without any arguments, flock defaults to be an exclusive lock (write lock) so I believe this is protected.

Reply:
It was just a news to me that /var/lib/columnstore/storagemanager is shared across nodes.

Reply:
https://mdbcdt.com/DOCS-2085/deploy/community-single-columnstore/#post-installation